src/coreclr/src/gc/gcinterface.h

```diff
@@ -782,7 +779,7 @@ class IGCHeap {
     virtual Object* AllocAlign8(gc_alloc_context* acontext, size_t size, uint32_t flags) = 0;
```

Comment:
Do we really need AllocAlign8 ?

Reply:
It is substantially different from regular Alloc and needed only on certain platforms. It seems convenient to have implementation in separate methods.

Reply:
For LOH - we could be forced to use LOH by the caller or decide to use it ourselves based on size. So Alloc has to know about large objects anyways.

Reply:
The implementation can be inside a separate method in the GC. That's fine.

I am looking at it from the GC interface definition point of view. We do have `GC_ALLOC_ALIGN8` flag. It is odd that you can call regular `Alloc` for all other combination flags, except when you have this specific flag - you need to call this other method.

Reply:
I see that `GC_ALLOC_ALIGN8` is not used. Perhaps it was meant to be used in Alloc.

Let's unify Alloc and AllocAlign8 at API level.

Reply:
> I see that GC_ALLOC_ALIGN8 is not used. Perhaps it was meant to be used in Alloc.

It has been used in .NET Native / CoreRT that has things more unified, but this unification did not make it into the master GC sources.

Reply:
Removed `AllocAlign8`. It did reduce some duplication.